### PR TITLE
Add: Bulk export patterns action.

### PR DESCRIPTION
**Pull request by jorgefilipecosta**
This PR adds the ability to bulk export a set of patterns.
If more than one pattern is selected to be exported a zip file with all the patterns is generated.

## How?
I selected multiple patterns.
Went to bulk edit and pressed export.
I downloaded the generated zip file, extracted it, and verified it contained the selected patterns as json files.

### CODE REVIEW

**Review by oandregal — COMMENTED** [review]
This is nice, though I found a bug with patterns that have the same name. This is how to reproduce:

- duplicate an existing pattern twice
- bulk export

I expected to have the 3 patterns in the ZIP but I only have 2 of them.

From what I see, the pattern's name is not unique, so we need to take that into account for the bulk export action.

**Comment by ntsekouras** [comment]
>I expected to have the 3 patterns in the ZIP but I only have 2 of them.

That's a good catch! I also made a quick research about alternatives to `client-zip` library, but it seems it fits well for our needs and is quite small.


Thanks Jorge!

**Comment by jorgefilipecosta** [comment]
> This is nice, though I found a bug with patterns that have the same name. This is how to reproduce:
> 
> * duplicate an existing pattern twice
> * bulk export
> 
> I expected to have the 3 patterns in the ZIP but I only have 2 of them.
> 
> From what I see, the pattern's name is not unique, so we need to take that into account for the bulk export action.

Good catch @oandregal, it was fixed 👍 

> I also made a quick research about alternatives to client-zip library, but it seems it fits well for our needs and is quite small.

Yes, I also did an analysis and client-zip seems to have a good tradeoff between providing what we need and having a small size.

File: packages/edit-site/src/components/page-patterns/dataviews-pattern-actions.js [hunk 1]
```diff
@@ -41,21 +42,49 @@ const { useHistory } = unlock( routerPrivateApis );
...
+			);
+		}
+		const nameCount = {};
+		const filesToZip = items.map( ( item ) => {
+			const name = kebabCase( item.title || item.name );
+			nameCount[ name ] = ( nameCount[ name ] || 0 ) + 1;
+			return {
+				name: `${
+					name +
+					( nameCount[ name ] > 1 ? '-' + nameCount[ name ] : '' )
```
Comment (ntsekouras) [inline packages/edit-site/src/components/page-patterns/dataviews-pattern-actions.js]: Probably it's better to follow the convention operating systems have and start from `1` for the same name files. 
For example:
```
copy.json
copy-1.json
```

So we should `nameCount[ name ] - 1`. What do you think?

**Review by ntsekouras — APPROVED** [review]
Besides the small name change in case of duplicate names, this LGTM. Thanks!

I'll pre approve and with a green CI, let's land.

File: packages/edit-site/src/components/page-patterns/dataviews-pattern-actions.js (hunk 1 above)
Comment (jorgefilipecosta) [reply packages/edit-site/src/components/page-patterns/dataviews-pattern-actions.js]: I agree with the suggestion, it was applied 👍

**Comment by paaljoachim** [comment]
Can one for instance import the exported patterns to a new site?
(I have not tested the export feature before.)

**Pull request merged**